src/store/GlobalState.js

```diff
@@ -0,0 +1,155 @@
+import { createContext, useEffect, useReducer } from 'react';
+import { collection, doc, addDoc, deleteDoc, updateDoc, query, onSnapshot } from 'firebase/firestore';
+
+import db from '../firebase';
+import appReducer from './appReducer';
+import { getCurrentMonth, getUKFormattedDate } from '../utils/helpers';
+
+const currentMonth = getCurrentMonth();
+
+const initialState = {
+  expenses: [],
+  selectedMonthExpenses: [],
+  currentMonth: currentMonth,
+  selectedMonth: currentMonth,
+  lastAddedExpense: {},
+  expenseToDelete: {},
+  expenseToEdit: {},
+  isConfirmMonthModalOpen: false,
+  isConfirmDeleteModalOpen: false,
+  isEditExpenseModalOpen: false,
```

Comment:
tracking open modals globally smells funny to me. maybe we can have a container which surrounds the possible modals and that one tracks which one is open. maybe something like:

```js
// pseudo code
ModalContainer.openAndCloseOthers(monthModal, param1, param2);
ModalContainer.closeAll();
```